lib/vagrant-cloudstack/action/run_instance.rb

```diff
@@ -251,6 +241,34 @@ def call(env)
           @app.call(env)
         end
 
+        def store_password(env, domain_config, job_id)
+          server_job_result = env[:cloudstack_compute].query_async_job_result({:jobid => job_id})
+          if server_job_result.nil?
+            env[:ui].warn(' -- Failed to retrieve job_result for retrieving the password')
+            return
+          end
+
+          while true
```

Comment:
The default VM timeout in vagrant is 300 seconds, but that is for the full deployment period.
I think that is too much for a single API call, maybe 10 seconds? 
Not really sure what make sense. I'll be happy with whatever you find reasonable.